docs/examples/ingestion/document_management_pipeline.ipynb

```diff
@@ -0,0 +1,272 @@
+{
```

Comment:
<p>might I know which cases hash could change</p><p>is document content change will change the hash</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/9135/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Reply:
Yea, the hash is set when a document/node is created. You can see here
https://github.com/run-llama/llama_index/blob/a98bdb8ecee513dc2e880f56674e7fd157d1dc3a/llama_index/schema.py#L386

Based on content + metadata

Reply:
does metadata contains timestamp?
if timestamp changes but content remain the same. should we reindex?

Reply:
metadata does contain a timestamp, at least for files loaded with SimpleDirectoryReader

Yea that's a good point. Room to optimize that later :)